driver-core/src/main/com/mongodb/operation/MixedBulkWriteOperation.java

```diff
@@ -284,14 +284,16 @@ private BulkWriteResult executeBulkWriteBatch(final WriteBinding binding, final
                 }
                 throw e;
             }
-        } else if (!(exception instanceof MongoWriteConcernWithResponseException)
-                && !shouldAttemptToRetry(originalBatch.getRetryWrites(), exception)) {
-            if (originalBatch.getRetryWrites()) {
-                logUnableToRetry(originalBatch.getPayload().getPayloadType().toString(), exception);
-            }
-            throw exception;
         } else {
```

Comment:
Please revert.  Generally best to avoid changing a file in a commit when it's not really relevant to the task at hand.

Reply:
Reverted.